Merge Symfony 7 into 2.0

loic425 opened the pull request:
| Q               | A
|-----------------|-----
| Branch?         | 2.0 
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets |
| License         | MIT

<!--
 - Bug fixes must be submitted against the 1.12 or 1.13 branches
 - Features and deprecations must be submitted against the 1.14 branch
 - Features, removing deprecations and BC breaks must be submitted against the 2.0 branch
 - Make sure that the correct base branch is set

 To be sure you are not breaking any Backward Compatibilities, check the documentation:
 https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

loic425 committed:
Add SF7 on CI matrix for packages

loic425 committed:
[SF7] Addressing component

loic425 committed:
[SF7] Add Symfony 7 support for some components (#16371)

| Q               | A
|-----------------|-----
| Branch?         | symfony-7
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

<!--
 - Bug fixes must be submitted against the 1.12 or 1.13 branches
 - Features and deprecations must be submitted against the 1.14 branch
- Features, removing deprecations and BC breaks must be submitted
against the 2.0 branch
 - Make sure that the correct base branch is set

To be sure you are not breaking any Backward Compatibilities, check the
documentation:

https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

After

![image](https://github.com/Sylius/Sylius/assets/8329789/1d00c77f-c9d5-4782-95c2-d426f149afd8)

loic425 committed:
[SF7] Add Symfony 7 support for Taxonomy bundle

loic425 committed:
Update src/Sylius/Bundle/TaxonomyBundle/composer.json

Co-authored-by: Dmitri Perunov <diimpp@gmail.com>

loic425 committed:
[SF7] Add Symfony 7 support for Taxonomy bundle (#16372)

| Q               | A
|-----------------|-----
| Branch?         | symfony-7
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

<!--
 - Bug fixes must be submitted against the 1.12 or 1.13 branches
 - Features and deprecations must be submitted against the 1.14 branch
- Features, removing deprecations and BC breaks must be submitted
against the 2.0 branch
 - Make sure that the correct base branch is set

To be sure you are not breaking any Backward Compatibilities, check the
documentation:

https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

After

![image](https://github.com/Sylius/Sylius/assets/8329789/9cf5d437-f290-4c24-a1ac-cd310f1f85f4)

loic425 committed:
[SF7] Add Symfony 7 support for Currency bundle

loic425 committed:
[SF7] Add Symfony 7 support for Inventory bundle

loic425 committed:
[SF7] Add Symfony 7 support for Locale bundle

loic425 committed:
[SF7] Add Symfony 7 support for Payment bundle

loic425 committed:
[SF7] Add Symfony 7 support for Attribute bundle

loic425 committed:
[SF7] Add Symfony 7 support for Attribute bundle (#16389)

| Q               | A
|-----------------|-----
| Branch?         | symfony-7
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets |
| License         | MIT

<!--
 - Bug fixes must be submitted against the 1.12 or 1.13 branches
 - Features and deprecations must be submitted against the 1.14 branch
- Features, removing deprecations and BC breaks must be submitted
against the 2.0 branch
 - Make sure that the correct base branch is set

To be sure you are not breaking any Backward Compatibilities, check the
documentation:

https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

After

![image](https://github.com/Sylius/Sylius/assets/8329789/7d562bcd-8237-4d2a-889d-84baf3bb3989)

loic425 committed:
[SF7] Add Symfony 7 support for Currency bundle (#16382)

| Q               | A
|-----------------|-----
| Branch?         | symfony-7
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no/
| Deprecations?   | no
| Related tickets | 
| License         | MIT

Based on #16372

<!--
 - Bug fixes must be submitted against the 1.12 or 1.13 branches
 - Features and deprecations must be submitted against the 1.14 branch
- Features, removing deprecations and BC breaks must be submitted
against the 2.0 branch
 - Make sure that the correct base branch is set

To be sure you are not breaking any Backward Compatibilities, check the
documentation:

https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

After

![image](https://github.com/Sylius/Sylius/assets/8329789/d512dc43-b79c-4501-8785-652c1584bac3)

loic425 committed:
[SF7] Add Symfony 7 support for Customer bundle

loic425 committed:
[SF7] Add Symfony 7 support for Inventory bundle (#16385)

| Q               | A
|-----------------|-----
| Branch?         | symfony-7
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets |
| License         | MIT

<!--
 - Bug fixes must be submitted against the 1.12 or 1.13 branches
 - Features and deprecations must be submitted against the 1.14 branch
- Features, removing deprecations and BC breaks must be submitted
against the 2.0 branch
 - Make sure that the correct base branch is set

To be sure you are not breaking any Backward Compatibilities, check the
documentation:

https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

Based on #16384

After

![image](https://github.com/Sylius/Sylius/assets/8329789/958ad8ee-a185-4857-b65a-40afe9603986)

loic425 committed:
[SF7] Add Symfony 7 support for Customer bundle (#16384)

| Q               | A
|-----------------|-----
| Branch?         | symfony-7
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

<!--
 - Bug fixes must be submitted against the 1.12 or 1.13 branches
 - Features and deprecations must be submitted against the 1.14 branch
- Features, removing deprecations and BC breaks must be submitted
against the 2.0 branch
 - Make sure that the correct base branch is set

To be sure you are not breaking any Backward Compatibilities, check the
documentation:

https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

After

![image](https://github.com/Sylius/Sylius/assets/8329789/4b067ce5-9616-4d4d-88e2-832e06c4a203)

loic425 committed:
[SF7] Add Symfony 7 support for Locale bundle (#16386)

| Q               | A
|-----------------|-----
| Branch?         | symfony-7
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

<!--
 - Bug fixes must be submitted against the 1.12 or 1.13 branches
 - Features and deprecations must be submitted against the 1.14 branch
- Features, removing deprecations and BC breaks must be submitted
against the 2.0 branch
 - Make sure that the correct base branch is set

To be sure you are not breaking any Backward Compatibilities, check the
documentation:

https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

After

![image](https://github.com/Sylius/Sylius/assets/8329789/903f7e56-906d-4fec-82ed-64dd327c6eab)

loic425 committed:
[SF7] Add Symfony 7 support for Payment bundle (#16388)

| Q               | A
|-----------------|-----
| Branch?         | symfony-7
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

<!--
 - Bug fixes must be submitted against the 1.12 or 1.13 branches
 - Features and deprecations must be submitted against the 1.14 branch
- Features, removing deprecations and BC breaks must be submitted
against the 2.0 branch
 - Make sure that the correct base branch is set

To be sure you are not breaking any Backward Compatibilities, check the
documentation:

https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

After

![image](https://github.com/Sylius/Sylius/assets/8329789/ec119d47-fb0b-466a-981a-585bec5b69f2)

loic425 committed:
[SF7] Add Symfony 7 support for Product bundle

loic425 committed:
[SF7] Add Symfony 7 support for Channel bundle

loic425 committed:
[SF7] Add Symfony 7 support for Channel bundle (#16400)

| Q               | A
|-----------------|-----
| Branch?         | symfony-7
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

<!--
 - Bug fixes must be submitted against the 1.12 or 1.13 branches
 - Features and deprecations must be submitted against the 1.14 branch
- Features, removing deprecations and BC breaks must be submitted
against the 2.0 branch
 - Make sure that the correct base branch is set

To be sure you are not breaking any Backward Compatibilities, check the
documentation:

https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

After

![image](https://github.com/Sylius/Sylius/assets/8329789/a8142003-9b0b-415a-a277-ea9d177384df)

loic425 committed:
[SF7] Add Symfony 7 support for Product bundle (#16399)

| Q               | A
|-----------------|-----
| Branch?         | symfony-7
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

<!--
 - Bug fixes must be submitted against the 1.12 or 1.13 branches
 - Features and deprecations must be submitted against the 1.14 branch
- Features, removing deprecations and BC breaks must be submitted
against the 2.0 branch
 - Make sure that the correct base branch is set

To be sure you are not breaking any Backward Compatibilities, check the
documentation:

https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

After

![image](https://github.com/Sylius/Sylius/assets/8329789/7be1aa41-9312-45ca-ae78-135507797816)

loic425 committed:
[SF7] Add Symfony 7 support for Taxation bundle

loic425 committed:
[SF7] Add Symfony 7 support for User bundle

loic425 committed:
[SF7] Add Symfony 7 support for Taxation bundle (#16403)

| Q               | A
|-----------------|-----
| Branch?         | symfony-7
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets |
| License         | MIT

<!--
 - Bug fixes must be submitted against the 1.12 or 1.13 branches
 - Features and deprecations must be submitted against the 1.14 branch
- Features, removing deprecations and BC breaks must be submitted
against the 2.0 branch
 - Make sure that the correct base branch is set

To be sure you are not breaking any Backward Compatibilities, check the
documentation:

https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

After

![image](https://github.com/Sylius/Sylius/assets/8329789/5cad92d4-efa8-4bdc-824d-baeaa101b0c8)

loic425 committed:
Apply suggestions from code review

loic425 committed:
[SF7] Add Symfony 7 support for Review bundle

loic425 committed:
[SF7] Add Symfony 7 support for Ui bundle

loic425 committed:
[SF7] Add Symfony 7 support for User bundle

loic425 committed:
[SF7] Add Symfony 7 support for Addressing bundle

GSadee committed:
Merge branch '2.0' into symfony-7

* 2.0: (212 commits)
  Change application's version to v1.13.4-dev
  Generate changelog for v1.13.3
  Change application's version to v1.13.3
  Change application's version to v1.12.19-dev
  Generate changelog for v1.12.18
  Change application's version to v1.12.18
  [CS][DX] Refactor
  Minor fixes after PR review
  Fix products grid for mobile
  [Admin][Taxon][Behat] Separate TreeElement from CreatePage
  [Admin][Taxon] Split tree to more hooks
  Fix phpstan
  [Admin][Taxon] Enable behat tests
  [Admin][Taxon] Remove old script
  [Admin] Taxon index
  Update PR template
  [Documentation] Update Sylius 1.12 version on the release cycle page
  Add upmerge from 2.0 to symfony-7 branch
  Minor changes
  Skip not working api scenarios
  ...

GSadee committed:
Merge branch '2.0' into symfony-7

* 2.0:
  [Admin] Fix slug generation button's icon
  Add info message to the account credentials section
  move logic from CustomerType to subscriber
  Enable tests
  Remove CustomerTypeExtension, as the whole logic is only used in the Admin
  Fix customer create/update
  Fixes after CR
  Minor fixes
  Fix styles
  Fix behats tags
  Make NewOrders component live
  Fix behats
  Change channel select button
  Refactor and fixes
  [Admin] Add a preliminary dashboard statistics implementation
  [Admin] Add dashboard component improvements
  [Behat] Configure dashboard defined elements
  [Behat] Enable dashboard statistics scenarios
  Change ship button styling

GSadee committed:
Merge branch '2.0' into symfony-7

* 2.0: (22 commits)
  [CS][DX] Refactor
  Fix error template finders specs and the way logged in admin user is checked
  [Behat] Fix """unexistent"""
  [Shop][Admin] Fix error pages location
  Slight cleanup
  Fix the build and improve a bit the failed step erros messages
  Move Error renderer to UiBundle
  Redirect to login page if the reset password token is wrong
  Decouple TwigErrorRenderer from AdminBundle
  Add tests
  Exclude ErrorRenderer and FlattenException from composer require checker
  Separate exact templates for Admin and Shop
  Separate the rendering of the error templates for Shop and Admin
  Create and add a 404 page for the Admin Panel using twig hooks
  Customer create edit page fixes
  Remove deprecated statistics
  Use interface for a response checker dependency
  [Admin][Taxon] Minor improvements
  [Admin][Taxon] Add info about component inside template
  [Admin][Taxon] Remove unnecessary lines
  ...

loic425 committed:
[UPMERGE] 2.0 -> symfony-7 (#16525)

This PR has been generated automatically.
For more details see
[upmerge_pr.yaml](/Sylius/Sylius/blob/1.13/.github/workflows/upmerge_pr.yaml).

**Remember!** The upmerge should always be merged with using `Merge pull
request` button.

In case of conflicts, please resolve them manually with usign the
following commands:
```
git fetch upstream
gh pr checkout <this-pr-number>
git merge upstream/symfony-7 -m "Resolve conflicts between 2.0 and symfony-7"
```

If you use other name for the upstream remote, please replace `upstream`
with the name of your remote pointing to the `Sylius/Sylius` repository.

Once the conflicts are resolved, please run `git merge --continue` and
push the changes to this PR.

loic425 committed:
[UPMERGE] 2.0 -> symfony-7 (#16535)

This PR has been generated automatically.
For more details see
[upmerge_pr.yaml](/Sylius/Sylius/blob/1.13/.github/workflows/upmerge_pr.yaml).

**Remember!** The upmerge should always be merged with using `Merge pull
request` button.

In case of conflicts, please resolve them manually with usign the
following commands:
```
git fetch upstream
gh pr checkout <this-pr-number>
git merge upstream/symfony-7 -m "Resolve conflicts between 2.0 and symfony-7"
```

If you use other name for the upstream remote, please replace `upstream`
with the name of your remote pointing to the `Sylius/Sylius` repository.

Once the conflicts are resolved, please run `git merge --continue` and
push the changes to this PR.

GSadee committed:
Merge branch '2.0' into symfony-7

* 2.0: (44 commits)
  [Behat] Add tests for filtering and sorting customer groups
  [Behat] Enable customer group browsing and deleting scenarios
  [Behat] Remove redundant at sign
  UserDropdownComponent - fix return type
  Remove unnecessary function
  [Admin][Locale] Fix path to form's template
  [Admin][Locale] Minor improvements
  [Admin][Locale] Add filtering tests
  [Admin][Locale] Remove update action
  [Admin][Locale] Enable behat tests
  [Admin][Locale] Add create/update page
  [Core] Remove LocaleTypeExtension
  [Admin] Change the name of the service to provide the logged in admin user
  [Admin] Extract logged in admin user providing
  [Admin] Fix 404 page when there is no route for url
  [Admin] Remove icon from existing avatar
  Fix missing template
  Fixes after CR
  [Admin] Add translation
  [Admin] Add twig hook, refactor behats
  ...

GSadee committed:
Merge branch '2.0' into symfony-7

* 2.0:
  Bump Twig Hooks to 0.3.x
  Disable sortable name field
  [Admin][Dashboard] Fix dropdown button

loic425 committed:
[UPMERGE] 2.0 -> symfony-7 (#16563)

This PR has been generated automatically.
For more details see
[upmerge_pr.yaml](/Sylius/Sylius/blob/1.13/.github/workflows/upmerge_pr.yaml).

**Remember!** The upmerge should always be merged with using `Merge pull
request` button.

In case of conflicts, please resolve them manually with usign the
following commands:
```
git fetch upstream
gh pr checkout <this-pr-number>
git merge upstream/symfony-7 -m "Resolve conflicts between 2.0 and symfony-7"
```

If you use other name for the upstream remote, please replace `upstream`
with the name of your remote pointing to the `Sylius/Sylius` repository.

Once the conflicts are resolved, please run `git merge --continue` and
push the changes to this PR.

loic425 committed:
[UPMERGE] 2.0 -> symfony-7 (#16578)

This PR has been generated automatically.
For more details see
[upmerge_pr.yaml](/Sylius/Sylius/blob/1.13/.github/workflows/upmerge_pr.yaml).

**Remember!** The upmerge should always be merged with using `Merge pull
request` button.

In case of conflicts, please resolve them manually with usign the
following commands:
```
git fetch upstream
gh pr checkout <this-pr-number>
git merge upstream/symfony-7 -m "Resolve conflicts between 2.0 and symfony-7"
```

If you use other name for the upstream remote, please replace `upstream`
with the name of your remote pointing to the `Sylius/Sylius` repository.

Once the conflicts are resolved, please run `git merge --continue` and
push the changes to this PR.

GSadee committed:
Merge branch '2.0' into symfony-7

* 2.0: (113 commits)
  [Admin] Rename parameter
  [Admin] Change command handlers and console command ids to dotted notation
  [Core] Change command handlers and console command ids to dotted notation
  [Core] Use new services
  [Core] Rename Message to Command
  [Admin] Rename Message to Command
  [Promotion] Remove deprecated commands
  [Ui] Remove deprecated commands
  [User] Remove deprecated commands
  [Admin] Configure and use anonymous view in store component
  [Admin] Add missing order notes information
  [Behat] Use test attribute to retrieve order notes
  [Admin] Render pagination and enabled filter test attributes
  [Behat] Enable scenarios and improve implementation for filtered pages redirections
  [Admin] Render view in store button in the variant edit page
  [CS][DX] Refactor
  [Behat] Improve view product in store action
  [Behat] Enable viewing product in store from variant edit page scenarios
  [Behat] Enable deleting multiple product variants scenario
  [Behat] Enable remaining managing users scenario
  ...

Wojdylak committed:
[UPMERGE] 2.0 -> symfony-7 (#16688)

This PR has been generated automatically.
For more details see
[upmerge_pr.yaml](/Sylius/Sylius/blob/1.13/.github/workflows/upmerge_pr.yaml).

**Remember!** The upmerge should always be merged with using `Merge pull
request` button.

In case of conflicts, please resolve them manually with usign the
following commands:
```
git fetch upstream
gh pr checkout <this-pr-number>
git merge upstream/symfony-7 -m "Resolve conflicts between 2.0 and symfony-7"
```

If you use other name for the upstream remote, please replace `upstream`
with the name of your remote pointing to the `Sylius/Sylius` repository.

Once the conflicts are resolved, please run `git merge --continue` and
push the changes to this PR.

GSadee committed:
Merge branch '2.0' into symfony-7

* 2.0:
  [API] Fix translation.name filters
  [Core] Fix #16603 SyliusCollector do not display API tag properly
  [Documentation] Update the dates on the release cycle page

Rafikooo committed:
[UPMERGE] 2.0 -> symfony-7 (#16723)

This PR has been generated automatically.
For more details see
[upmerge_pr.yaml](/Sylius/Sylius/blob/1.13/.github/workflows/upmerge_pr.yaml).

**Remember!** The upmerge should always be merged with using `Merge pull
request` button.

In case of conflicts, please resolve them manually with usign the
following commands:
```
git fetch upstream
gh pr checkout <this-pr-number>
git merge upstream/symfony-7 -m "Resolve conflicts between 2.0 and symfony-7"
```

If you use other name for the upstream remote, please replace `upstream`
with the name of your remote pointing to the `Sylius/Sylius` repository.

Once the conflicts are resolved, please run `git merge --continue` and
push the changes to this PR.

loic425 committed:
[UPMERGE] 2.0 -> symfony-7 (#16725)

This PR has been generated automatically.
For more details see
[upmerge_pr.yaml](/Sylius/Sylius/blob/1.13/.github/workflows/upmerge_pr.yaml).

**Remember!** The upmerge should always be merged with using `Merge pull
request` button.

In case of conflicts, please resolve them manually with usign the
following commands:
```
git fetch upstream
gh pr checkout <this-pr-number>
git merge upstream/symfony-7 -m "Resolve conflicts between 2.0 and symfony-7"
```

If you use other name for the upstream remote, please replace `upstream`
with the name of your remote pointing to the `Sylius/Sylius` repository.

Once the conflicts are resolved, please run `git merge --continue` and
push the changes to this PR.

loic425 committed:
[SF7] Add Symfony 7 support for Money bundle

GSadee committed:
[SF7][Money] Remove unneeded templating helpers

loic425 committed:
[UPMERGE] 2.0 -> symfony-7 (#16761)

This PR has been generated automatically.
For more details see
[upmerge_pr.yaml](/Sylius/Sylius/blob/1.13/.github/workflows/upmerge_pr.yaml).

**Remember!** The upmerge should always be merged with using `Merge pull
request` button.

In case of conflicts, please resolve them manually with usign the
following commands:
```
git fetch upstream
gh pr checkout <this-pr-number>
git merge upstream/symfony-7 -m "Resolve conflicts between 2.0 and symfony-7"
```

If you use other name for the upstream remote, please replace `upstream`
with the name of your remote pointing to the `Sylius/Sylius` repository.

Once the conflicts are resolved, please run `git merge --continue` and
push the changes to this PR.

GSadee committed:
[SF7][Money] Add a note about constructors changes in the UPGRADE file

GSadee committed:
[SF7][User] Fix serialized token in spec for LoggedInAdminUserProvider

GSadee committed:
Merge branch '2.0' into symfony-7

* 2.0:
  [Money] Deprecate unneeded templating helpers
  [CS][DX] Refactor
  [CS][DX] Refactor

NoResponseMate committed:
minor #16757 [SF7] Add Symfony 7 support for MoneyBundle (GSadee)

This PR was merged into the symfony-7 branch.

Discussion
----------

| Q               | A
|-----------------|-----
| Branch?         | symfony-7
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no<!-- don't forget to update the UPGRADE-*.md file -->
| Related tickets | replaces #16387
| License         | MIT

<!--
 - Bug fixes must be submitted against the 1.13 branch
 - Features and deprecations must be submitted against the 1.14 branch
 - Features, removing deprecations and BC breaks must be submitted against the 2.0 branch
 - Make sure that the correct base branch is set

 To be sure you are not breaking any Backward Compatibilities, check the documentation:
 https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->


Commits
-------
  [SF7] Add Symfony 7 support for Money bundle
  [SF7][Money] Remove unneeded templating helpers
  [SF7][Money] Add a note about constructors changes in the UPGRADE file

GSadee committed:
[Review] Fix adding PHPUnit result cache to gitignore file

GSadee committed:
Merge branch '2.0' into symfony-7

* 2.0:
  [API] Minor improvements for ProductVariantNormalizer
  [API] Remove deprecated Sylius 1.13 tests

loic425 committed:
[SF7] Add Symfony 7 support for Shipping bundle

loic425 committed:
Sort dependencies alphabetically

loic425 committed:
[SF7] Add Symfony 7 support for Order bundle

GSadee committed:
[SF7] Add Symfony 7 support for Promotion Bundle

GSadee committed:
[Composer] Upgrade Webpack to ^2.1

loic425 committed:
[SF7] Add Symfony 7 support for Payum bundle

loic425 committed:
[Composer] Upgrade Webpack to ^2.1 (#16781)

| Q               | A
|-----------------|-----
| Branch?         | symfony-7
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations? | no<!-- don't forget to update the UPGRADE-*.md file
-->
| Related tickets | 
| License         | MIT

Upgrade Webpack to ^2.1 so that its version is consistent across all
packages
<!--
 - Bug fixes must be submitted against the 1.13 branch
 - Features and deprecations must be submitted against the 1.14 branch
- Features, removing deprecations and BC breaks must be submitted
against the 2.0 branch
 - Make sure that the correct base branch is set

To be sure you are not breaking any Backward Compatibilities, check the
documentation:

https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

loic425 committed:
[SF7] Add Symfony 7 support for Promotion Bundle (#16777)

| Q               | A
|-----------------|-----
| Branch?         | symfony-7
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations? | no<!-- don't forget to update the UPGRADE-*.md file
-->
| Related tickets | mentioned in
https://github.com/Sylius/Sylius/issues/15481
| License         | MIT

<!--
 - Bug fixes must be submitted against the 1.13 branch
 - Features and deprecations must be submitted against the 1.14 branch
- Features, removing deprecations and BC breaks must be submitted
against the 2.0 branch
 - Make sure that the correct base branch is set

To be sure you are not breaking any Backward Compatibilities, check the
documentation:

https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

loic425 committed:
[SF7] Add Symfony 7 support for Core bundle

loic425 committed:
Rework install database command

GSadee committed:
[SF7][Core] Refactor console commands in Core Bundle

GSadee committed:
[SF7][Core] Bump up some dev dependencies

GSadee committed:
[SF7][Core] Additional refactor console commands after PR review

GSadee committed:
[SF7][Core] Revert changing Symfony require in composer

GSadee committed:
[Core] Fix static analysis

loic425 committed:
[SF7] Add Symfony 7 support for State machine abstraction

GSadee committed:
[Core] Fix Behat installer tests

GSadee committed:
[Core][Behat] Remove scenarios for removed show available plugins CLI command

GSadee committed:
[Core][CLI] Remove newly added and unused methods from CommandExecutor

loic425 committed:
[SF7] Add Symfony 7 support for Core bundle

loic425 committed:
[SF7] Add Symfony 7 support for Admin bundle

NoResponseMate committed:
minor #16798 [SF7] Add Symfony 7 support for Core bundle (loic425)

This PR was merged into the symfony-7 branch.

Discussion
----------

| Q               | A
|-----------------|-----
| Branch?         | symfony-7
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

<!--
 - Bug fixes must be submitted against the 1.13 branch
 - Features and deprecations must be submitted against the 1.14 branch
 - Features, removing deprecations and BC breaks must be submitted against the 2.0 branch
 - Make sure that the correct base branch is set

 To be sure you are not breaking any Backward Compatibilities, check the documentation:
 https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

![image](https://github.com/user-attachments/assets/f4c7dce8-ed15-4a68-aa90-53ad07fd5d74)


Commits
-------
  [SF7] Add Symfony 7 support for Core bundle

loic425 committed:
[SF7] Add Symfony 7 support for Shop bundle

loic425 committed:
Fix PHPStan error

NoResponseMate committed:
minor #16800 [SF7] Add Symfony 7 support for Admin bundle (loic425)

This PR was merged into the symfony-7 branch.

Discussion
----------

| Q               | A
|-----------------|-----
| Branch?         | symfony-7
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

<!--
 - Bug fixes must be submitted against the 1.13 branch
 - Features and deprecations must be submitted against the 1.14 branch
 - Features, removing deprecations and BC breaks must be submitted against the 2.0 branch
 - Make sure that the correct base branch is set

 To be sure you are not breaking any Backward Compatibilities, check the documentation:
 https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->


Commits
-------
  [SF7] Add Symfony 7 support for Admin bundle

loic425 committed:
[UPMERGE] 2.0 -> symfony-7 (#16786)

This PR has been generated automatically.
For more details see
[upmerge_pr.yaml](/Sylius/Sylius/blob/1.13/.github/workflows/upmerge_pr.yaml).

**Remember!** The upmerge should always be merged with using `Merge pull
request` button.

In case of conflicts, please resolve them manually with usign the
following commands:
```
git fetch upstream
gh pr checkout <this-pr-number>
git merge upstream/symfony-7 -m "Resolve conflicts between 2.0 and symfony-7"
```

If you use other name for the upstream remote, please replace `upstream`
with the name of your remote pointing to the `Sylius/Sylius` repository.

Once the conflicts are resolved, please run `git merge --continue` and
push the changes to this PR.

NoResponseMate committed:
minor #16801 [SF7] Add Symfony 7 support for Shop bundle (loic425)

This PR was merged into the symfony-7 branch.

Discussion
----------

| Q               | A
|-----------------|-----
| Branch?         | symfony-7
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

<!--
 - Bug fixes must be submitted against the 1.13 branch
 - Features and deprecations must be submitted against the 1.14 branch
 - Features, removing deprecations and BC breaks must be submitted against the 2.0 branch
 - Make sure that the correct base branch is set

 To be sure you are not breaking any Backward Compatibilities, check the documentation:
 https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

![image](https://github.com/user-attachments/assets/5929eed1-77cc-46ef-b293-0b327ba9415f)


Commits
-------
  [SF7] Add Symfony 7 support for Shop bundle
  Fix PHPStan error

GSadee committed:
Merge branch '2.0' into symfony-7

* 2.0:
  [Maintenance] Remove message buses duplications
  [CS][DX] Refactor
  Minor CS fix in workflow
  [Resource] Add validation for translatable interface with new namespace
  Apply suggestions from code review
  Both resource and grid unstable versions
  Try SyliusGridBundle 1.13
  [Core] Move validation of translation classes to Core

GSadee committed:
Merge branch '2.0' into symfony-7

* 2.0:
  [CS][DX] Refactor
  [CS][DX] Refactor
  [CS][DX] Refactor
  [CI] Update Sylius branches in refactor workflow
  [Behat] Switch another random JS failed scenarios to Chromedriver
  [Behat] Switch another random JS failed scenarios to Chromedriver
  [Behat] Switch random JS failed scenarios to Chromedriver
  [PHPStan] Temporarily ignore errors with ResourceBundle 1.11 to fix static analysis
  Minor improvements in PHPStan configuration
  [Taxonomy] Fix Taxon spec
  [PHPStan] Support both sylius/resource-bundle 1.10 and 1.11
  [CS][DX] Refactor

GSadee committed:
Merge branch '2.0' into symfony-7

* 2.0:
  Fix data grid sorting when there are no defaults
  [Admin][CatalogPromotion] Fix icon for show variants link

GSadee committed:
Resolve conflicts between 2.0 and symfony-7

GSadee committed:
[Addressing] Upgrade matthiasnoback/symfony-dependency-injection-test dependency

GSadee committed:
[UPMERGE] 2.0 -> symfony-7 (#16851)

This PR has been generated automatically.
For more details see
[upmerge_pr.yaml](/Sylius/Sylius/blob/1.13/.github/workflows/upmerge_pr.yaml).

**Remember!** The upmerge should always be merged with using `Merge pull
request` button.

In case of conflicts, please resolve them manually with usign the
following commands:
```
git fetch upstream
gh pr checkout <this-pr-number>
git merge upstream/symfony-7 -m "Resolve conflicts between 2.0 and symfony-7"
```

If you use other name for the upstream remote, please replace `upstream`
with the name of your remote pointing to the `Sylius/Sylius` repository.

Once the conflicts are resolved, please run `git merge --continue` and
push the changes to this PR.

GSadee committed:
Merge branch '2.0' into symfony-7

* 2.0:
  [CS][DX] Refactor
  [Unit][Admin] Add product variant collection filtering test
  [UpgradeAPI] Add a note about PostgreSQLDriverExceptionListener removal
  [Upgrade] Fix indentation
  [API] Remove obsolete PostgreSQLDriverExceptionListener
  Remove api-platform-3 and add payment-request to upmerge workflow

loic425 committed:
[SF7] Add Symfony 7 support for API bundle

loic425 committed:
Bump phparkitect version to 0.3

loic425 committed:
Replace master request with main request

loic425 committed:
Adapt message handlers for SF7

GSadee committed:
[SF7] Adapt normalizers and denormalizers in ApiBundle

GSadee committed:
[SF7] Adjust PHPSpecs of normalizers in ApiBundle

GSadee committed:
[SF7] Remove deprecated interfaces from normalizers and denormalizers in ApiBundle

GSadee committed:
[SF7] Adapt supportsNormalization method of normalizers in ApiBundle

GSadee committed:
[SF7] Adjust PHPSpecs in ApiBundle

GSadee committed:
[SF7] Disable cache for normalizers and denormalizers to avoid infinite loop

GSadee committed:
[SF7] Fix PHPSpec for ProductAttributeValueNormalizer

GSadee committed:
[ApiBundle] Extract AsMessageHandler check in PHPSpec

NoResponseMate committed:
minor #16864 [SF7] Add Symfony 7 support for API bundle (loic425)

This PR was merged into the symfony-7 branch.

Discussion
----------

| Q               | A
|-----------------|-----
| Branch?         | symfony-7
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

<!--
 - Bug fixes must be submitted against the 1.13 branch
 - Features and deprecations must be submitted against the 1.14 branch
 - Features, removing deprecations and BC breaks must be submitted against the 2.0 branch
 - Make sure that the correct base branch is set

 To be sure you are not breaking any Backward Compatibilities, check the documentation:
 https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->


Commits
-------
  [SF7] Add Symfony 7 support for API bundle
  Bump phparkitect version to 0.3
  Replace master request with main request
  Adapt message handlers for SF7
  [SF7] Adapt normalizers and denormalizers in ApiBundle
  [SF7] Adjust PHPSpecs of normalizers in ApiBundle
  [SF7] Remove deprecated interfaces from normalizers and denormalizers in ApiBundle
  [SF7] Adapt supportsNormalization method of normalizers in ApiBundle
  [SF7] Adjust PHPSpecs in ApiBundle
  [SF7] Disable cache for normalizers and denormalizers to avoid infinite loop
  [SF7] Fix PHPSpec for ProductAttributeValueNormalizer

GSadee committed:
[ApiBundle] Fix namespaces of MessageHandlerAttributeTrait in PHPSpec

NoResponseMate committed:
minor #16875 [ApiBundle] Extract AsMessageHandler check in PHPSpec (GSadee)

This PR was merged into the symfony-7 branch.

Discussion
----------

| Q               | A
|-----------------|-----
| Branch?         | symfony-7
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | based on #16864 
| License         | MIT

<!--
 - Bug fixes must be submitted against the 1.13 branch
 - Features and deprecations must be submitted against the 1.14 branch
 - Features, removing deprecations and BC breaks must be submitted against the 2.0 branch
 - Make sure that the correct base branch is set

 To be sure you are not breaking any Backward Compatibilities, check the documentation:
 https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->


Commits
-------
  [ApiBundle] Extract AsMessageHandler check in PHPSpec
  [ApiBundle] Fix namespaces of MessageHandlerAttributeTrait in PHPSpec

loic425 committed:
[SF7] Uniform package versions

loic425 committed:
Fix phparkitect config

loic425 committed:
Give it a try without phparkitekt

NoResponseMate committed:
[Maintenance] Fix phparkitect issues

GSadee committed:
Merge branch '2.0' into symfony-7

* 2.0: (26 commits)
  Update namespaces to Resource ^1.11 in 2.0
  [CS][DX] Refactor
  [CS][DX] Refactor
  [Maintenance] Update resource changes comparison in the upgrade file
  [Maintenance] Fix LazyOption repository template covariants
  [Maintenance] Note resource namespaces changes
  [Docs] Update namespaces
  [Maintenance] Update leftover namespaces
  Cover the scenario for allowing access for only correctly logged in users
  Enable the remaining cart suites
  [Maintenance] Fix translations' duplicated validation
  [Api][Unit] Fix it_gets_all_product_variants_when_invalid_product_filter_provided test
  [Api] Replace sylius.api to sylius_api
  [Api] Cover sorting and filtration of resources
  [Behat] Refine codebase for promotion and cart contexts
  Enable dashboard and homepage suites
  Minor Product config improvement
  Enable the rest of product tests and correct order of the attribute values in the response
  Correct contract tests and behat
  Bring back non exposed ProductOptionValue
  ...

GSadee committed:
Merge branch '2.0' into symfony-7

* 2.0:
  [Behat] Disable some customer scenarios
  [API] Split state provider configurations into admin and shop sections
  [Core] Add new payment repository method
  [Behat] Change the term status to state in payment and shipment scenarios
  [Behat] Refine codebase for shipment and payment contexts
  [Unit] Add new shipment and payment tests and refine existing ones
  [Behat] Adjust shipment and payment URLs
  [API][Shop] Secure shipment and payment endpoints
  [API][Shop] Move shipment and payment endpoints under the order resource
  [Unit] Add disableShopContext util function
  [Behat] Add requestGet method to API client
  [Unit] Change nAWw2jewpA token value to token
  [Unit] Add new response util functions
  [Unit] Refine order placer trait
  [Behat][API] Enable customer suite
  [Behat] Enable customer scenarios

jakubtobiasz commented on src/Sylius/Bundle/ApiBundle/CommandHandler/Account/ChangeShopUserPasswordHandler.php:
```diff
@@ -17,10 +17,11 @@
 use Webmozart\Assert\Assert;
 
-final readonly class ChangeShopUserPasswordHandler implements MessageHandlerInterface
+#[AsMessageHandler]
+final readonly class ChangeShopUserPasswordHandler
```
It's unnecessary, as it has no effect. It's used by autoconfiguration which is disabled for bundles.

loic425 replied:
Yeah indeed, but Symfony uses it on Symfony\Component\Cache\Messenger\EarlyExpirationHandler.
Maybe it's just an information.

BTW, AsCommand attribute is now required, maybe the'll force using it at a moment.

loic425 committed:
[SF7] Upgrade matrix to tests Symfony 7

loic425 committed:
Fix templating version

GSadee committed:
[CI] Upgrade matrix to test SF7 and PHP 8.3

GSadee committed:
[Composer] Upgrade babdev/pagerfanta-bundle dependency

GSadee committed:
[Composer] Temporarily upgrade sylius/mailer-bundle dependency to alpha version

GSadee committed:
[Composer] Upgrade doctrine/event-manager dependency

GSadee committed:
[Composer] Upgrade knplabs/knp-gaufrette-bundle dependency

GSadee committed:
[Composer] Remove sylius-labs/polyfill-symfony-framework-bundle dependency

GSadee committed:
[Composer] Remove sylius-labs/polyfill-symfony-security dependency

GSadee committed:
[Composer] Remove sylius-labs/polyfill-symfony-event-dispatcher dependency

GSadee committed:
[Composer] Temporarily upgrade sylius/grid-bundle dependency to alpha version

GSadee committed:
[Composer] Temporarily upgrade sylius/resource-bundle dependency to alpha version

GSadee committed:
[Composer] Upgrade consolidation/log and psr/log dependencies

GSadee committed:
[Composer] Try to remove conflict to doctrine/orm

GSadee committed:
[Composer] Remove willdurand/hateoas and willdurand/hateoas-bundle dependencies

GSadee committed:
[SF7] Update configs

loic425 committed:
Fix RequestMatcher usage

GSadee committed:
Merge branch '2.0' into symfony-7

* 2.0:
  [CS][DX] Refactor
  [UserBundle] Add Jms/Serializer-bundle
  [UserBundle] FOS/Rest
  [CS][DX] Refactor

GSadee committed:
Merge branch '2.0' into symfony-7

* 2.0:
  [Docs] Fix link in grids customization guide

GSadee committed:
Fix allowExtraFields validation option in PayumBundle

GSadee committed:
Fix invalid argument of CommandDenormalizer

GSadee committed:
Fix declaration of ResourceToIdentifierTransformer::transform method

GSadee committed:
Remove using container in old migrations

GSadee committed:
Fix static analysis

GSadee committed:
Update composer-require-checker whitelist

GSadee committed:
Remove CircularDependencyBreakingErrorListener

GSadee committed:
Fix declaration of AppKernel::registerContainerConfiguration method

GSadee committed:
Merge branch '2.0' into symfony-7

* 2.0:
  ShopUserBasedExtension - rename function name
  [Behat] Enable shop checkout context in address book suite
  [Behat] Enable customer address book scenarios
  Refactor a bit the validation of Statistics, enable unit test
  [API] Cover missing suite for getting errors in admin's language
  [Api] Fix spec tests
  [Api] Enable behat tests
  [Api] Add change password operation
  [Api] Secure the customer get endpoint

GSadee committed:
[Composer] Update SF require to ^7.0

GSadee committed:
Use MessageHandlerAttributeTrait in ChangeShopUserPasswordHandlerSpec

GSadee committed:
[API] Change the order of customers in contract tests response

GSadee committed:
Remove PHPUnits related with removed hateos dependency

GSadee committed:
Refactor LocaleUsageChecker to use translation classes instead of interfaces

mpysiak committed:
Fix behat errors

mpysiak committed:
Change JS tests order (for now)

# Conflicts:
#	.github/workflows/ci_e2e-mysql.yaml

GSadee committed:
Revert refactor of LocaleUsageChecker after fix on ResourceBundle

GSadee committed:
Fix UserInterface by removing nonexistent interface

GSadee committed:
[Composer] Change symfony/templating require versions

mpysiak committed:
Change event to preRemove (for now)

mpysiak committed:
Change phpstan ignore

mpysiak committed:
Fix event args type

mpysiak committed:
fix MoneyBundle/composer.json

mpysiak committed:
fix specs

GSadee committed:
Merge branch '2.0' into symfony-7

* 2.0:
  [CI] Split JS builds for Panther and Chromedriver + add additional rerun
  Revert "[CI] Upmerge Action False Positive"
  Remove note about twig/twig from CONFLICTS.md file
  Bump twig packages
  [CI] Complete upmerge action successfully even if there are no changes between branches
  Add missing tags and cover scenarios in shopping_cart suite
  [CS][DX] Refactor

GSadee committed:
[Composer] Fix syntax error in LocaleBundle

GSadee committed:
Update composer-require-checker whitelist

mpysiak committed:
Install guzzle7 client

mpysiak committed:
Revert "Remove CircularDependencyBreakingErrorListener"

This reverts commit 0aeacd97e1206c39e1cf636079a31d0aab510ffa.

mpysiak committed:
fix CircularDependencyBreakingErrorListener

mpysiak committed:
Bump postgresql version in matrix

mpysiak committed:
Revert "[API] Change the order of customers in contract tests response"

This reverts commit 81a1fae989c5126bc13be88b8bd5736a4dba23d2.

mpysiak committed:
Change fixtures - add createdAt different time for customers

mpysiak committed:
Remove unused use

mpysiak committed:
Bump postgres in matrix

GSadee committed:
Resolve conflicts between 2.0 and symfony-7

GSadee committed:
[CI] Fix matrix.json file

GSadee committed:
[Composer] Add doctrine/doctrine-bundle to bundles

GSadee committed:
[UPMERGE] 2.0 -> symfony-7 (#16950)

This PR has been generated automatically.
For more details see
[upmerge_pr.yaml](/Sylius/Sylius/blob/1.13/.github/workflows/upmerge_pr.yaml).

**Remember!** The upmerge should always be merged with using `Merge pull
request` button.

In case of conflicts, please resolve them manually with usign the
following commands:
```
git fetch upstream
gh pr checkout <this-pr-number>
git merge upstream/symfony-7 -m "Resolve conflicts between 2.0 and symfony-7"
```

If you use other name for the upstream remote, please replace `upstream`
with the name of your remote pointing to the `Sylius/Sylius` repository.

Once the conflicts are resolved, please run `git merge --continue` and
push the changes to this PR.

mpysiak commented on src/Sylius/Bundle/ApiBundle/composer.json:
```diff
@@ -54,6 +54,9 @@
             "dev-main": "2.0-dev"
+        },
+        "symfony": {
+            "require": "7.0.*"
```
```suggestion
            "require": "^7.0"
```

In every bundle

GSadee replied:
I would be even for bumping the minimum version to ^7.1, but definitely in a separate PR after merge this one 🖖🏻
